### ISSUE TITLE
[Bug]: group by doesn't get any search result

### ISSUE THREAD

**Issue by shanghaikid** [milvus-io/milvus#29852]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): node sdk
```

### Current Behavior

group by doesn't get any search result.

1. create a collection with schema
```javascript
const params: any = {
    collection_name: collectionName,
    consistency_level: ConsistencyLevelEnum.Strong,
    fields: [
      {
        name: VECTOR_FIELD_NAME,
        description: 'Vector field',
        data_type: vectorType,
        dim: Number(dim),
      },
      {
        name: 'id',
        description: 'ID field',
        data_type: DataType.Int64,
        is_primary_key: true,
        autoID,
      },
      {
        name: 'int64',
        description: 'int64 field',
        data_type: 'Int64',
      },
      {
        name: 'float',
        description: 'Float field',
        data_type: DataType.Float,
      },
      {
        name: 'bool',
        description: 'bool field',
        data_type: DataType.Bool,
      },
      {
        name: 'default_value',
        // default_value: DEFAULT_VALUE,
        description: 'int64 field',
        data_type: 'Int64', // test string data type
      },
      {
        name: 'varChar',
        description: 'VarChar field',
        data_type: DataType.VarChar,
        max_length: MAX_LENGTH,
        is_partition_key: partitionKeyEnabled,
      },
      {
        name: 'json',
        description: 'JSON field',
        data_type: DataType.JSON,
      },
      {
        name: 'int32_array',
        description: 'int array field',
        data_type: DataType.Array,
        element_type: 'Int32', // test string element type
        max_capacity: maxCapacity || MAX_CAPACITY,
      },
      {
        name: 'float_array',
        description: 'int array field',
        data_type: DataType.Array,
        element_type: DataType.Float,
        max_capacity: maxCapacity || MAX_CAPACITY,
      },
      {
        name: 'varChar_array',
        description: 'varChar array field',
        data_type: DataType.Array,
        element_type: DataType.VarChar,
        max_capacity: maxCapacity || MAX_CAPACITY,
        max_length: MAX_LENGTH,
      },
    ],
    enable_dynamic_field: true
  };
```
2.  insert data
3.  flush data
4.  create index
5.  load collection
6.  search
```javascript
    const searchWithData = await milvusClient.search({
      collection_name: COLLECTION_NAME,
      filter: '',
      data: [1, 2, 3, 4],
      limit: limit,
      group_by_field: 'varChar',
    });
```

Always get no search result. 
```javascript
 {
      status: {
        error_code: 'Success',
        reason: 'search result is empty',
        code: 0,
        retriable: false,
        detail: ''
      },
      results: []
    }
```

I can get search result if I don't set `group_by_field`

**Comment by MrPresent-Han** [comment]
this is because the flush interface of milvus called by node js is not a sync method, resulting in the searched_segments are growing segments, not flushed segments. Whereas search_group_by doesn't support search_growing segments currently. So this is a known defect and will be fixed when growing segments supporting search_group_by

**Comment by yanliang567** [comment]
keep it

**Comment by xiaofan-luan** [comment]
@yanliang567 

should already be fixed. please verify on this

**Comment by yanliang567** [comment]
/assign @shanghaikid 
please help to verify the fix on latest master build.
/unassign @MrPresent-Han

**Comment by shanghaikid** [comment]
fixed, verified on `master-20240306-a88c8967`
